File: api_sc_ate_condicionais/alunos/DavydLucas/nenem.py
def main():
    #Entrada

    nome_aluno = input("Digite o nome do aluno: ")
    nota_redacao = int(input("Digite a nota da Redação do aluno: "))
    P1 = int(input("Digite o peso da Redação(P1): "))
    nota_linguagens = int(input("Digite a nota de Linguagens do aluno: "))
    P2 = int(input("Digite o peso da prova de Linguagens(P2): "))
    nota_matematica = int(input("Digite a nota de Matemática do aluno: "))
    P3 = int(input("Digite o peso da prova de Matemática(P3): "))
    nota_humanas = int(input("Digite a nota de Humanas do aluno: "))
    P4 = int(input("Digite o peso da prova de Humanas(P4): "))
    nota_natureza = int(input("Digite a nota de Natureza do aluno: "))
    P5 = int(input("Digite o peso da prova de Natureza do aluno(P5): "))

    #Processamento

    #Cálculo dos pontos

    pontos_linguagens = calcular_pontos(nota_linguagens)
    pontos_matematica = calcular_pontos(nota_matematica)
    pontos_humanas = calcular_pontos(nota_humanas)
    pontos_natureza = calcular_pontos(nota_natureza)

    #Cálculo da média simples
    media_simples = round(calcular_media_simples(nota_redacao, nota_linguagens, nota_matematica, nota_humanas, nota_natureza))

    #Cálculo da média pondera
    media_ponderada = round(calcular_media_pondera(nota_redacao, nota_linguagens, nota_matematica, nota_humanas, nota_natureza, P1, P2, P3, P4, P5))

    #Classificação
    nota_final = media_ponderada

    if nota_final > 0 and nota_final < 450:
        classificacao = "INSUFICIENTE"
        observacao = "Impedido de ir para Universidade. Deverá fazer mais um ano de ensino de reforço"
    elif nota_final >= 450  and nota_final < 650:
        classificacao = "SUFICIENTE"
        observacao = "Poderá conseguir uma vafa. Porém em Curso/Universidade com baixa procura (nota de corte baixa)"
    elif nota_final >= 650 and nota_final < 750:
        classificacao = "MUITO BOM"
        observacao = "Poderá pleitear vaga em bons cursos/Universidades.(Nota de corte alta)"
    else:
        classificacao = "EXCELENTE"
        observacao = "Poderá escolher qualquer curso em qualquer universidade de Serigate"

    #Saída

    print(f"************* SERIGATE NENEM **************")
    print(f"Aluno           : {nome_aluno}")
    print(f"-------------------------------------------")
    print(f"Redação         : {nota_redacao} (peso: {P1})")
    print(f"Linguagens      : {nota_linguagens} (peso: {P2})")
    print(f"Matemática      : {nota_matematica} (peso: {P3})")
    print(f"Humanas         : {nota_humanas} (peso: {P4})")
    print(f"Natureza        : {nota_natureza} (peso: {P5})")
    print(f"-------------------------------------------")
    print(f"Média Simples   : {media_simples}\n")
    print(f">>>>>>>>>>>>>>>> RESULTADO <<<<<<<<<<<<<<<<")
    print(f"Média Ponderada: {media_ponderada}")
    print(f"Classificação: {classificacao}")
    print(f"Observação: {observacao}")
    print(f"-------------------------------------------")


def calcular_media_pondera(A1, A2, A3, A4, A5, P1, P2, P3, P4, P5):
    valor_media_ponderada = ((A1 * P1) + (A2 * P2) + (A3 * P3) + (A4 * P4) + (A5 * P5))/(P1 + P2 + P3 + P4 + P5)
    return valor_media_ponderada

def calcular_media_simples(A1, A2, A3, A4, A5):
    valor_media_simples = (A1 + A2 + A3 + A4 + A5)/5
    return valor_media_simples

def calcular_pontos(nota):
    pontos = (nota * 1000)/45
    return pontos


main()

File: api_sc_ate_condicionais/alunos/Ronaldo/nenem.js
import {question} from 'readline-sync'

function main(){
    //Entrada
    const nome = (question('Digite o seu nome: '))
    const questoes_linguagens = Number(question('Digite quantas questoes acertou na prova linguagens: '))
    const peso_linguagens = Number(question('Digite o peso da nota de linguagens: '))
    const questoes_humanas = Number(question('Digite quantas questoes acertou na prova de humanas: '))
    const peso_humanas = Number(question('Digite o peso da nota de humanas: '))
    const questoes_matematica = Number(question('Digite quantas questoes acertou na prova de matematica: '))
    const peso_matematica = Number(question('Digite o peso da nota de matematica: '))
    const questoes_natureza = Number(question('Digite quantas questoes acertou na prova de natureza: '))
    const peso_natureza = Number(question('Digite o peso da nota de natureza: '))
    const nota_redacao = Number(question('Digite a nota da redacao: '))
    const peso_nota_redacao = Number(question('Digite o peso da nota da redacao: '))

    //Processamento
    const nota_linguagens = ((questoes_linguagens / 45) * 1000)
    const nota_humanas = ((questoes_humanas / 45) * 1000)
    const nota_matematica = ((questoes_matematica / 45) * 1000)
    const nota_natureza = ((questoes_natureza / 45) * 1000)
    
    const media_simples = ((nota_linguagens + nota_humanas + nota_matematica + nota_natureza + nota_redacao) / 5)
    const media_ponderada = calcular_media_ponderada(nota_linguagens, peso_linguagens, nota_humanas, peso_humanas, nota_matematica, peso_matematica, nota_natureza, peso_natureza, nota_redacao, peso_nota_redacao)
    const classificacao = verificar_classificacao(media_ponderada)
    const observacao = verificar_observacao(media_ponderada)

    //Saída
    console.log(`************** SERIGATE NENEM **************`)
    console.log(`Aluno          : ${nome}`)
    console.log(`____________________________________________`)
    console.log(`Redação        : ${nota_redacao.toFixed(2)}     Peso: ${peso_nota_redacao}`)
    console.log(`Linguagens     : ${nota_linguagens.toFixed(2)}     Peso: ${peso_linguagens}`)
    console.log(`Matemática     : ${nota_matematica.toFixed(2)}     Peso: ${peso_matematica}`)
    console.log(`Humanas        : ${nota_humanas.toFixed(2)}     Peso: ${peso_humanas}`)
    console.log(`Natureza       : ${nota_natureza.toFixed(2)}     Peso: ${peso_natureza}`)
    console.log(`____________________________________________`)
    console.log(`Média Simples  : ${media_simples.toFixed(2)}`)
    console.log(`>>>>>>>>>>>>>>>>> RESULTADO <<<<<<<<<<<<<<<<<`)
    console.log(`Média Ponderada: ${media_ponderada.toFixed(2)}`)
    console.log(`Classificação  : ${classificacao}`)
    console.log(`Observação     : ${observacao}`)
    console.log(`____________________________________________`)

}

function calcular_media_ponderada(nota_linguagens, peso_linguagens, nota_humanas, peso_humanas, nota_matematica, peso_matematica, nota_natureza, peso_natureza, nota_redacao, peso_nota_redacao){
    const resultado = (((peso_linguagens*nota_linguagens) + (peso_humanas*nota_humanas) + (peso_matematica*nota_matematica) + (peso_natureza*nota_natureza) + (peso_nota_redacao*nota_redacao)) / (peso_linguagens + peso_humanas + peso_natureza + peso_matematica + peso_nota_redacao))
    return resultado
}

function verificar_classificacao(media_ponderada){
    if(media_ponderada > 0 && media_ponderada < 450){
        return 'INSUFICIENTE!'
    }else if(media_ponderada > 450 && media_ponderada < 650){
        return 'SUFICIENTE!'
    }else if(media_ponderada > 650 && media_ponderada < 750){
        return 'MUITO BOM!'
    }else if(media_ponderada > 750 && media_ponderada < 1000){
        return 'EXCELENTE!'
    }else{
        return 'A NOTA DIGITADA É INVÁLIDA!'
    }
}

function verificar_observacao(media_ponderada){
    if(media_ponderada > 0 && media_ponderada < 450){
        return 'Impedido de ir para a Universidade. Deverá fazer mais um ano de ensino de reforço!'
    }else if(media_ponderada > 450 && media_ponderada < 650){
        return 'Poderá conseguir uma vaga. Porém em Curso/Universidade com baixa procura(nota de corte baixa.)!'
    }else if(media_ponderada > 650 && media_ponderada < 750){
        return 'Poderá pleitear vaga em bons Cursos/Universidades(nota de corte alta.)!'
    }else if(media_ponderada > 750 && media_ponderada < 1000){
        return 'Poderá escolher qualquer curso em qualquer universidade de Serigate!'
    }else{
        return 'A NOTA DIGITADA É INVÁLIDA!'
    }
}
main()